lib/internal/errors.js

```diff
@@ -858,6 +858,44 @@ const genericNodeError = hideStackFrames(function genericNodeError(message, erro
   return err;
 });
 
+/**
+ * Determine the specific type of a value for type-mismatch errors.
+ * @param {*} value
+ * @returns {string}
+ */
+function determineSpecificType(value) {
+  let type = '';
+
+  if (value == null) {
+    type += value;
+  } else if (typeof value === 'function' && value.name) {
+    type = `function ${value.name}`;
+  } else if (typeof value === 'object') {
+    if (value.constructor?.name) {
+      type = `an instance of ${value.constructor.name}`;
+    } else {
+      const inspected = lazyInternalUtilInspect()
+        .inspect(value, { depth: -1 });
+
+      if (StringPrototypeIncludes(inspected, '[Object: null prototype]')) {
+        type = 'an instance of Object';
+      } else {
+        type = inspected;
+      }
+    }
+  } else {
+    let inspected = lazyInternalUtilInspect()
+      .inspect(value, { colors: false });
+    if (inspected.length > 25) {
```

Comment:
If the length is 28, it's not longer to display the full string than slice it at 25 chars and add 3 dots after that.
```suggestion
    if (inspected.length > 28) {
```

Reply:
Good point. It was there before, and I tried to make as few updates as necessary.